org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/NullReferenceTest.java

```diff
@@ -18568,4 +18568,33 @@ void m(String packageName, IPS[] packages) {
 			"""
 		});
 }
+// https://github.com/eclipse-jdt/eclipse.jdt.core/issues/1461
+public void testGH1461() {
+	if (this.complianceLevel == 0) {
+		buildAllCompliancesTestSuite(testClass());
+		this.complianceLevel = ClassFileConstants.JDK1_8;
+	}
+	if (this.complianceLevel < ClassFileConstants.JDK1_7) return;
+	runNegativeTest(
+		new String[] {
+			"X.java",
+			"import java.util.Objects;\n"
+			+ "public class X {\n"
+			+ "	public void test() { \n"
+			+ "		String name = null;\n"
+			+ "		if (Objects.isNull(name)) { \n"
+			+ "			System.out.println(\"Name is null\");\n"
+			+ "			return;\n"
+			+ "		}\n"
+			+ "		System.out.println(name.substring(0, 4));\n"
+			+ "	}\n"
+			+ "}\n"
```

Comment:
Meanwhile we can use text blocks for source could within tests, which avoids all the boiler plate of concatenating multi-line strings.

Reply:
Fixed.